### PR TITLE
Fix & optimize System Statement permission & Statement existence checks

### PR DESCRIPTION
**Pull request by Dav1dV**
~Depends on https://github.com/GovReady/govready-q/pull/1468~

### CODE REVIEW

File: controls/views.py [hunk 1]
```diff
@@ -921,25 +921,23 @@ def import_component(request):
...
+    """Raise an HTTP 404 if the user doesn't have the statement system permission"""
+
+    # Allow access if the user has superuser or staff member permissions respectively
+    if not (request.user.is_superuser or request.user.is_staff):
+
+        # Allow access if the user has the statement system permission
+        system = get_object_or_404(System, root_element=statement.consumer_element)
+        raise_404_if_system_permissions_not(request, system, system_permission,
+            lambda permission, subaction, user_message:
+                log_system_object_action(request, system,
```
Comment (davidpofo) [inline controls/views.py:934]: add this to PR

File: controls/views.py [hunk 2]
```diff
@@ -138,6 +138,47 @@ def control(request, catalog_key, cl_id):
...
+        if not_callback:
+            not_callback(permission, 'permission_denied',
+                         'You do not have permission to change this system')
+
+        raise Http404(f'Access denied:  You do not have permission')
+        # 404 instead of 403 + vague message  to not leak info.
+        #   Message only conveyed when DEBUG True
+
+
+def log_system_action(request, system, action, message={}):
```
Comment (Dav1dV) [inline controls/views.py:154]: Note:  Used in a parallel PR in progress ...

Just didn't spend time on combining only to revert in that later PR ; )

File: controls/views.py (hunk 1 above)
Comment (Dav1dV) [reply controls/views.py:1057]: @davidpofo Made this independent of #1468

File: controls/views.py [hunk 3]
```diff
@@ -1003,25 +1044,24 @@ def import_component(request):
...
+        # Allow access if the user has the statement system permission
+        system = get_object_or_404(System, root_element=statement.consumer_element)
+        raise_404_if_system_permissions_not(request, system, system_permission,
+            lambda permission, subaction, user_message:
+                log_system_object_action(request, system,
+                                         [permission, action, subaction],
+                                         'statement', 'id', statement.id)
+        )
+
+def raise_404_if_statement_permission_not_change(request, statement, action):
```
Comment (davidpofo) [inline controls/views.py:1062]: need a doc string

**Review by davidpofo — COMMENTED** [review]
@Dav1dV looks good in practice but because of its nested nature, this requires automated tests. Could you draft some unit tests for each function introduced in this PR. The tests should live in controls/tests.py

**Comment by davidpofo** [comment]
Not needed

**Pull request closed without merging**